src/dune_engine/print_diff.ml

```diff
@@ -22,16 +22,17 @@ let print ?(skip_trailing_cr = Sys.win32) path1 path2 =
       ]
   in
   let normal_diff () =
-    let path, args, skip_trailing_cr_arg =
+    let path, args, skip_trailing_cr_arg, file_arg =
       let which prog = Bin.which ~path:(Env.path Env.initial) prog in
       match which "git" with
       | Some path ->
         ( path
         , [ "diff"; "--no-index"; "--color=always"; "-u" ]
-        , "--ignore-cr-at-eol" )
+        , "--ignore-cr-at-eol"
+        , Unix.readlink )
```

Comment:
I'm surprised if this works. `readlink` just returns a path stored in the symlink, without combining it with anything if it's relative.
I think the correct way to do it is something like `(fun f -> let l = Unix.readlink f in if Filename.is_relative l then Filename.dirname f ^/ l else l)`, but it needs testing.

Reply:
Good catch. I think I was looking for `realpath`. Strangely enough, it doesn't seem to exist in the stdlib and I had to implement something like it.

Reply:
Any reason you're converting the paths to absolute?
Don't we prefer relative paths where possible? And here it's clearly possible.

Reply:
Sorry, I didn't read the implementation of the function, just the name. I think the name should be changed because it doesn't do what `realpath` command or C function do.